xncml/core.py

```diff
@@ -278,3 +286,127 @@ def to_ncml(self, path=None):
         xml_output = xmltodict.unparse(self.ncroot, pretty=True)
         with open(path, 'w') as fd:
             fd.write(xml_output)
+
+    def to_cf_dict(self):
+        """Convert internal representation to a CF-JSON dictionary.
+
+        The CF-JSON specification includes `data` for variables, but this is not included here.
+
+        Returns
+        -------
+        Dictionary with `dimensions` and `variables` keys. May also optionally include an `attributes` key and a
+        `groups` key. Additional keys prefixed with `@` may be included for <netcdf> tag attributes,
+        for example `@location`.
+
+        References
+        ----------
+        http://cf-json.org/specification
+        """
+        res = OrderedDict()
+        nc = self.ncroot['netcdf']
+
+        for key, val in nc.items():
+            if key[0] == '@':
+                res[key] = val
+            if key == 'dimension':
+                res.update(_dims_to_json(val))
+            if key == 'group':
+                res.update(_groups_to_json(val))
+            if key == 'attribute':
+                res.update(_attributes_to_json(val))
+            if key == 'variable':
+                res.update(_variables_to_json(val))
+
+        return res
+
+
+def _dims_to_json(dims: list) -> dict:
+    """The dimensions object has dimension id:size as its key:value members."""
+    out = OrderedDict()
+    for dim in dims:
+        if int(dim['@length']) > 1:
+            out[dim['@name']] = int(dim['@length'])
+
+    return {'dimensions': out}
+
+
+def _groups_to_json(groups: list) -> dict:
+    out = OrderedDict()
+    for group in groups:
+        name = group['@name']
+        out[name] = OrderedDict()
+        if 'attribute' in group:
+            out[name].update(_attributes_to_json(group['attribute']))
+        if 'group' in group:
+            out[name].update(_groups_to_json(group['group']))
+
+    return {'groups': out}
+
+
+def _attributes_to_json(attrs: list) -> dict:
+    """The attributes object contains arbitrary attributes as its key:value members."""
+    SPECIAL_ATTRS = ['missing_value', 'cell_methods']
```

Comment:
Why are those two skipped?

The CF-json spec shows an example where missing values are translated to json's `null`, but I don't see this here, so why not use this attribute?

A comment to explain this would be nice at least.

Reply:
Again, taken from the repo above, which provides no explanation for the removal of these attributes. Not sure what to do with this. I could ask the metocean folks the reason for this.